tensorly/regression/cp_plsr.py

```diff
@@ -151,17 +181,40 @@ def fit(self, X, Y):
                 self.Y_factors[1], T.index[:, component], comp_Y_factors_1
             )
 
+            B = T.lstsq(self.X_factors[0], T.reshape(comp_Y_factors_0, (-1, 1)))[0]
+            self.coef_ = T.index_update(
+                self.coef_,
+                T.index[:, component],
+                T.reshape(B, (-1,)),
+            )
+
             # Deflation
-            X -= CPTensor(
-                (None, [T.reshape(ff, (-1, 1)) for ff in comp_X_factors])
-            ).to_tensor()
+            X -= cp_to_tensor((None, [T.reshape(ff, (-1, 1)) for ff in comp_X_factors]))
```

Comment:
Why not use tl.tenalg.outer? It's just a rank-1 CP, i.e. a single outer product of vectors. That may be faster as avoiding the checks and creation of None weights to handle.